miniconf/src/array.rs

```diff
@@ -93,30 +93,29 @@ macro_rules! depth {
         }
 
         impl<T: TreeAny<{$y - 1}>, const N: usize> TreeAny<$y> for [T; N] {
-            fn get_by_key<K>(&self, mut keys: K) -> Result<&dyn Any, Traversal>
+            fn ref_any_by_key<K>(&self, mut keys: K) -> Result<&dyn Any, Traversal>
             where
                 K: Keys,
             {
                 let item = get(self, &mut keys, false)?;
-                item.get_by_key(keys).map_err(Traversal::increment)
+                item.ref_any_by_key(keys).map_err(Traversal::increment)
             }
 
-            fn get_mut_by_key<K>(&mut self, mut keys: K) -> Result<&mut dyn Any, Traversal>
+            fn mut_any_by_key<K>(&mut self, mut keys: K) -> Result<&mut dyn Any, Traversal>
             where
                 K: Keys,
             {
                 let item = get_mut(self, &mut keys, false)?;
-                item.get_mut_by_key(keys).map_err(Traversal::increment)
+                item.mut_any_by_key(keys).map_err(Traversal::increment)
             }
         }
     )+}
 }
 depth!(2 3 4 5 6 7 8 9 10 11 12 13 14 15 16);
 
 impl<const N: usize, T> KeyLookup for [T; N] {
-    fn len() -> usize {
-        N
-    }
+    const LEN: usize = N;
+    const NAMES: &'static [&'static str] = &[];
```

Comment:
Whats the point of adding these to the keylookup trait? Not seeing the value of the `NAMES` const.

Reply:
It's cleaner. Otherwise in the derive macro we need to try to hide it in the inherent impl.